### DISCUSSION TITLE
Add Module Name to JGroups

### DISCUSSION THREAD

**Discussion by eitan-rosenberg** [belaban/JGroups#688] [Ideas]
Now days people are using modules.

So please have a look at this article:
http://branchandbound.net/blog/java/2017/12/automatic-module-name/ 

From the article:
"This post explains the first step you can take as a library maintainer on your way to becoming a Java module. This first step boils down to picking a module name, and adding it as  Automatic-Module-Name: -module name- entry to the library's MANIFEST.MF. That's it".

Regards.

**Comment by belaban** [comment]
So I added this to the POM:

----
```
<manifestEntries>
       <Automatic-Module-Name>org.jgroups</Automatic-Module-Name>
</manifestEntries>
```

Good?

**Comment by eitan-rosenberg** [comment]
Good morning,

The end result is something like this ( from Apache Commons IO commons-io-2.11.0.jar ) :

![image](https://user-images.githubusercontent.com/48506054/204208984-4824fefc-a375-4ec3-937b-2eb8f227e7a5.png)

If you want me to test is before you commit then send me a link please.

Thanks.

**Reply by belaban** [reply]
I already committed this, the result is what you pasted above: `Automatic-Module-Name: org.jgroups`

**Reply by eitan-rosenberg** [reply]
Ok, I will test it now.

**Reply by eitan-rosenberg** [reply]
The latest in maven is 5.2.10.Final

**Comment by tristantarrant** [comment]
You need to build from source

**Reply by eitan-rosenberg** [reply]
Thanks,
I will wait until the next Maven update.
Regards.

**Comment by belaban** [comment]
That will take a while... If you build a snapshot JAR locally and test, and there's still an issue, the fix (to be fixed later) will be in 5.2.11. If you wait for 5.2.11, and the issue still persists, then the fix will be in 5.2.12 only...

**Reply by eitan-rosenberg** [reply]
Ok,
Thanks.

**Comment by eitan-rosenberg** [comment]
Good morning,
The addition of : "Automatic-Module-Name: org.jgroups" seems to work, thank you.
Regards and Happy new year

**Reply by belaban** [reply]
Same to you, too!

**Comment by eitan-rosenberg** [comment]
Close
